async/kafka/kafka.go

```diff
@@ -194,6 +206,90 @@ func (c *consumer) Consume(ctx context.Context) (<-chan async.Message, <-chan er
 	return chMsg, chErr, nil
 }
 
+func consume(ctx context.Context, c *consumer) (<-chan async.Message, <-chan error, error) {
+
+	chMsg := make(chan async.Message, c.buffer)
+	chErr := make(chan error, c.buffer)
+
+	log.Infof("consuming messages from topic '%s' without using consumer group", c.topic)
+	pcs, err := c.partitions()
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "failed to get partitions")
+	}
+	// When kafka cluster is not fully initialized, we may get 0 partions.
+	if len(pcs) == 0 {
+		return nil, nil, errors.New("got 0 partitions")
+	}
+
+	for _, pc := range pcs {
```

Comment:
Optional: could we break the code in private function in order to be more readable?